cosmos/operators/local.py

```diff
@@ -471,14 +472,29 @@ def _handle_post_execution(self, tmp_project_dir: str, context: Context) -> None
             self.callback_args.update({"context": context})
             self.callback(tmp_project_dir, **self.callback_args)
 
+    def _delete_sql_files(self, tmp_project_dir: Path, resource_type: str) -> None:
+        dest_target_dir, dest_conn_id = self._configure_remote_target_path()
+        source_run_dir = Path(tmp_project_dir) / f"target/{resource_type}"
+        files = [str(file) for file in source_run_dir.rglob("*") if file.is_file()]
+        from airflow.io.path import ObjectStoragePath
+
+        for file_path in files:
+            dest_file_path = self._construct_dest_file_path(dest_target_dir, file_path, source_run_dir, resource_type)  # type: ignore
+            dest_object_storage_path = ObjectStoragePath(dest_file_path, conn_id=dest_conn_id)
+            dest_object_storage_path.unlink()
+            self.log.info("Copied %s to %s", file_path, dest_object_storage_path)
+
     def _handle_async_execution(self, tmp_project_dir: str, context: Context, async_context: dict[str, Any]) -> None:
```

Comment:
What is the benefit of having a same method to handle teardown and setup, and not separate methods?

Reply:
If please do let me know if you want me to refactor it, I can try to do it in a separate PR.